metric_learn/base_metric.py

```diff
@@ -589,6 +589,97 @@ def _validate_calibration_params(strategy='accuracy', min_rate=None,
                          'Got {} instead.'.format(type(beta)))
 
 
+class _TripletsClassifierMixin(BaseMetricLearner):
+  """Base class for triplets learners.
+  """
+
+  _tuple_size = 3  # number of points in a tuple, 3 for triplets
+
+  def predict(self, triplets):
+    """Predicts the ordering between sample distances in input triplets.
+
+    For each triplets, returns 1 if the first two elements are closer than the
+    first and last and -1 if not.
+
+    Parameters
+    ----------
+    triplets : array-like, shape=(n_triplets, 3, n_features) or (n_triplets, 3)
+      3D Array of triplets to predict, with each row corresponding to three
+      points, or 2D array of indices of triplets if the metric learner
+      uses a preprocessor.
+
+    Returns
+    -------
+    prediction : `numpy.ndarray` of floats, shape=(n_constraints,)
+      Predictions of the ordering of pairs, for each triplet.
+    """
+
+    # Aren't the following lines redundant as they are called on
+    # decision_function (as in quadruplets predict) ???
```

Comment:
Yes, we should leave the check out here and in the quadruplets case as well.

Reply:
I think it is better to leave the check in `decision_function`, as it can be called separately.

Reply:
Yes, this is indeed not needed, it looks like you can safely remove the calls to `check_is_fitted` and `check_input` here and in the quadruplet version